src/registry/controller.ts

```diff
@@ -0,0 +1,44 @@
+import { type SignerConfig } from "../helpers/connection.js";
```

Comment:
Controller based method on the registry class. This pattern is used throughout the SDK. We have the core functions, that take a config, and then these are pulling into the exposed classes.

Reply:
My goal with all of this is to hide all the glue code into helpers, making the actual work (like this) much easier to grok.